tests/main/presenters/test_search_presenters.py

```diff
@@ -680,3 +681,256 @@ def test_build_lots_and_categories_link_tree_with_lot_and_no_category_filters_ha
+                    flask.request,
+                    flask.request.args,
+                    g9_builder,
+                    'services',
+                    'g-cloud-9',
+                    Href(flask.url_for('.search_services')),
+                    self.search_api_client
+                )
+
+        assert tree == self.tree_with_cloud_software__option_2__option_2_1__selected
+        assert(
+            [(child['id'], child['selected']) for child in tree[0]['children'][0]['children'][0]['children']] ==
+            [('checkboxTreeExample-option-2.1', True), ('checkboxTreeExample-option-2.2', False)]
+        )
+
+        tree_child_value_lists = [[child.get('id') for child in node.get('children', [])] for node in tree]
+        tree_child_values = [item for sublist in tree_child_value_lists for item in sublist]
+
+        assert 'checkboxTreeExample-option-7.1' not in tree_child_values
+        assert tree_child_values.count('checkboxTreeExample-option-2.1') == 1
+
+    def test_build_lots_and_categories_link_tree_with_lot_and_filter_and_same_named_filter_in_tree_with_no_parent(self):
+        url = "/g-cloud/search?lot=cloud-software&checkboxTreeExample=option+2.1"
+
+        self.category_filter_group['filters'] += [
+            {
+                'label': 'Option 7',
+                'name': 'checkboxTreeExample',
+                'id': 'checkboxTreeExample-option-7',
+                'value': 'option 7',
+                'children': [
+                    {
+                        'label': 'Option 2.1',
+                        'name': 'checkboxTreeExample',
+                        'id': 'checkboxTreeExample-option-2.1',
+                        'value': 'option 2.1'
+                    }, {
+                        'label': 'Option 7.1',
+                        'name': 'checkboxTreeExample',
+                        'id': 'checkboxTreeExample-option-7.1',
+                        'value': 'option 7.1'
+                    }
+                ]
+            }
+        ]
+        with self.app.test_request_context(url):
+            tree = build_lots_and_categories_link_tree(
+                self.framework,
+                self.framework['lots'],
+                self.category_filter_group.copy(),
+                flask.request,
+                flask.request.args,
+                g9_builder,
+                'services',
+                'g-cloud-9',
+                Href(flask.url_for('.search_services')),
+                self.search_api_client
+            )
+
+        tree_child_value_selected_lists = [
+            [(child.get('id'), child['selected']) for child in node.get('children', [])] for node in tree
+        ]
+        tree_child_value_selected = [item for sublist in tree_child_value_selected_lists for item in sublist]
+
+        assert ('checkboxTreeExample-option-7.1', False) in tree_child_value_selected
+        assert tree_child_value_selected.count(('checkboxTreeExample-option-2.1', True)) == 2
```

Comment:
So if I understand this right, if there's no `parentCategory` and the child appears in two categories, we expand the categories for both of them?

Reply:
Yeah that's right! It's weird but I think it's a) the intended behaviour and b) actually the right thing to do. Because without a `parentCategory` we can't find our way back up the tree.

We can't just put selected for a random choice, because then we might assign selected to something the user didn't select.

We shouldn't do it for none because they selected something.

The weird thing about this PR, and this bug, is that it just propagates the `parentCatgory` url argument to where it needs to be (and also makes sure it gets stripped where necessary). The code to 'select' the correct categories was already there.

Some validation of the `parentCategory` vs the selection could be used to find a way around this, but I'm not sure there's a user need for it 🤔

Reply:
Makes sense - thanks!